src/validation/validation-middleware.ts

```diff
@@ -0,0 +1,36 @@
+import { Request, Response, NextFunction } from 'express';
+import { Schema, ValidationErrorItem } from 'joi';
+import { User } from '../routers/user-router';
+
+const errorResponse = (schemaErrors: ValidationErrorItem[]) => {
+    const errors = schemaErrors.map((error: ValidationErrorItem) => {
+        const { path, message } = error;
+        return { path, message };
+    });
+
+    return {
+        status: 'failed',
+        errors
+    };
+};
+
+export const validateSchema = (schema: Schema, users: { users: User[] }) =>
+    (req: Request, res: Response, next: NextFunction) => {
+        const { error } = schema.validate(req.body, {
+            abortEarly: false,
+            allowUnknown: false
+        });
+        const login = users.users.find((item: User) => item.login === req.body.login);
```

Comment:
I think it is bad idea to connect body validator with master data. That validate function should do only body check. Let's imagine your data stored in DB. You will get big problem with that realization

Reply:
added a separate check